experimental/oak_baremetal_channel/SPEC.MD

```diff
@@ -0,0 +1,293 @@
+# Oak Baremetal Communication Protocol Specification
```

Comment:
I think the spec is more general than Oak Baremetal, anyways we can move / rename later.